### PR TITLE
[PORT] pre-newfood cooking-palooza

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:

- https://github.com/tgstation/tgstation/pull/52771/
- https://github.com/tgstation/tgstation/pull/52527/
- https://github.com/tgstation/tgstation/pull/48817/
- https://github.com/tgstation/tgstation/pull/58110/
- https://github.com/tgstation/tgstation/pull/54573/

## About The Pull Request

Pigeon remarked that he was interested in trying to push the /tg/ newfood and cooking system at one point, but it would have to be heavily atomized to be done effectively. This PR seeks to port a non-exhaustive list of pre- and mid- newfood PRs from TG to more easily merge the rework and cooking system at a later point in time.

## Why It's Good For The Game

/tg/s food system is categorically superior and they have had numerous bug and feature fixes that haven't been taken advantage of in this codebase yet. This PR is made to serve as a stepping-stone to shrink the gap between the state where we are at now and our goal.

Some of these features may have been half-implemented already, but for some reason weren't fully realized. Such as the salami being in our food.dmi but never actually used as an icon by an item.

This PR will also make minor implementations of /tg/ "moth culture" as we are adjacent to their common-core

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

non-moths hate these cloth-filled things!
image

 
</details>

## Changelog
:cl: ShizCalev, ATHATH, OnlineGirlfriend, Iamgoofball, The0bserver, Stewydeadmike, TripleZeta and RKz
add: A special interaction for trying to put a mushroom on an Arnold pizza has been added.
add: Arnold pizza references its source material more accurately/often now.
add: Chefs can now cook moffins! Yaaay moffins! Check the pastries subsection of the crafting menu to make some! Cutting up clothes and bedsheets can get you started!
add: Adds 3 new plants, and 6 new recipes, as well as a juice reagent. Check out your local Megaseed Vendor today!
fix: Fixed Arnold pizza not checking for TRAIT_NODISMEMBER, resulting in a message being sent to the user even if they were fine after eating it.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/food/snacks_pizza.dm [hunk 1]
```diff
@@ -169,20 +169,25 @@
 	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pepperoni" = 2, "9 millimeter bullets" = 2)
 
 /obj/item/reagent_containers/food/snacks/proc/try_break_off(mob/living/M, mob/living/user) //maybe i give you a pizza maybe i break off your arm
```
Comment (itsmeow) [inline code/modules/food_and_drinks/food/snacks_pizza.dm:171]: the fuck are these procs? looks like some LRP BS, consider removing it?

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 2]
```diff
@@ -340,6 +340,31 @@
...
+
+/obj/item/reagent_containers/food/snacks/muffin/moffin/examine(mob/user)
+	if(!ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/item/organ/tongue/T = H.getorganslot(ORGAN_SLOT_TONGUE)
+
+		if(!T)
+			return
+
+		if(!HAS_TRAIT(H, TRAIT_AGEUSIA))
```
Comment (itsmeow) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: Aguesia is taste based, why would ageusia affect your ability to look at and discern what is on the food?

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/food/snacks_pizza.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:171]: wdym? arnold pizza?

File: code/modules/food_and_drinks/food/snacks_pastry.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/modules/food_and_drinks/food/snacks_pastry.dm]: i havent fixed that examine yet, its cannibalized code atm

File: code/modules/food_and_drinks/food/snacks_pizza.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:171]: removing that proc would be feature removal which is not within the scope of this pr.

Comment (itsmeow) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:171]: then there is no need to update the pizza within this PR either if the feature is to be removed

Comment (Tsar-Salat) [reply code/modules/food_and_drinks/food/snacks_pizza.dm:171]: removing that proc would be feature removal which is not within the scope of this pr.

File: code/modules/food_and_drinks/food/snacks_pastry.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/modules/food_and_drinks/food/snacks_pastry.dm]: dumb check anyway, i just removed it

**Comment by Tsar-Salat** [comment]
I didnt expect to be the one porting #8748 , back when I made this pr in november.

Thats how things shook out, so I will fix any conflicts with that pr.

**Comment by Tsar-Salat** [comment]
Whqt the fuck

**Comment by Tsar-Salat** [comment]
❤️

**Review by PowerfulBacon — APPROVED** [review]

**Review by itsmeow — APPROVED** [review]
It does a thing. I am amazed this PR still compiles

**Pull request merged**